exercises/practice/react/lib/react.ex

```diff
@@ -0,0 +1,42 @@
+defmodule React do
+  @doc """
+  Start a reactive system
+  """
+  @spec new(cells :: [{:input, String.t(), any} | {:output, String.t(), [String.t()], fun()}]) ::
+          {:ok, pid}
+  def new(cells) do
+  end
```

Comment:
You could create a type here to break down this complex type
```suggestion
  @type input :: {:input, name :: String.t(), value :: any}
  @type output ::
          {:output, name :: String.t(), dependencies :: list(String.t()), callback :: function()}
  @type cell_definition :: input() | output()

  @opaque cells :: pid

  @doc """
  Start a reactive system
  """
  @spec new(cell_definitions :: [cell_definition()]) :: {:ok, cells()}
  def new(cell_definitions) do
  end
```